prov/hook/perf/src/hook_perf.c

```diff
@@ -883,22 +885,51 @@ static int perf_hook_fabric(struct fi_fabric_attr *attr,
 	}
 
 	hook_fabric_init(&fab->fabric_hook, HOOK_PERF, attr->fabric, hprov,
-			 &perf_fabric_fid_ops);
+			 &perf_fabric_fid_ops, &perf_prov_ctx);
```

Comment:
I think we want to try replacing HOOK_PERF with a struct hook_provider * (now called struct hook_prov_ctx).